freestyle/shared/src/main/scala/freestyle/free.scala

```diff
@@ -117,30 +109,23 @@ object freeImpl {
           q"case class $Req[..$TTs](..$params) extends $OP[$Res]"
       }
 
-      def raiser: DefDef = {
-        val injected = {
-          /*filter: if !v.mods.hasFlag(Flag.IMPLICIT)*/
-          val args = params.map(_.name)
-          q"FreeS.inject[$OP, $LL]( $ReqC[..${tparams.map(_.name)} ](..$args) )"
-        }
-
-        val tpt = reqDef.tpt.asInstanceOf[AppliedTypeTree]
-        val (liftType, impl) = tpt match {
-          case tq"OpSeq[$aa]" => (tpt, q"FreeS.liftPar($injected)" )
-          case tq"OpPar[$aa]" => (tpt, injected )
-          case _ => // Note: due to filter in getRequestDefs, this case is unreachable.
-            fail(s"unknown abstract type found in @free container: $tpt : raw: ${showRaw(tpt)}")
-        }
+      def raiser: DefDef =
+        q"""
+          override def ${reqDef.name}[..$tparams](...${reqDef.vparamss}): ${reqDef.tpt} =
+            $inj( $ReqC[..${tparams.map(_.name)} ](..${params.map(_.name)}) )
```

Comment:
Note that, because `$inj` is appearing in the scope of the value parameters of the method, we need a fresh name; otherwise it may accidentally collide with any of those in `reqDef.vparamasss`.